lib/player.js

```diff
@@ -2406,6 +2421,33 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
       this.loadEventManager_.listen(
           mediaElement, 'timeupdate', onVideoProgress);
       this.onVideoProgress_();
+      if (this.manifest_.nextUrl) {
+        if (this.config_.streaming.minCompletionPreloadNextUrl > 0 &&
+            this.config_.streaming.minCompletionPreloadNextUrl < 100) {
+          const onTimeUpdate = async () => {
+            const completionRatio =
+                this.video_.currentTime / this.video_.duration;
+            if (!isNaN(completionRatio)) {
+              const percent = Math.round(100 * completionRatio);
+              const minCompletionPreloadNextUrl =
+                  this.config_.streaming.minCompletionPreloadNextUrl;
+              if (percent >= minCompletionPreloadNextUrl) {
+                this.loadEventManager_.unlisten(
+                    mediaElement, 'timeupdate', onTimeUpdate);
+                goog.asserts.assert(this.manifest_.nextUrl,
+                    'this.manifest_.nextUrl should be valid.');
+                this.preloadNextUrl_ =
+                    await this.preload(this.manifest_.nextUrl);
+              }
+            }
+          };
+          this.loadEventManager_.listen(
```

Comment:
Question: does this event manager handle properly removing the event listener when the player is destroyed?

Reply:
Yes, that is managed in unload.